system/helper/functions.php

```diff
@@ -327,15 +327,22 @@ function deserialize($varValue, $blnForceArray=false)
 		return $blnForceArray ? array() : '';
 	}
 
-	// Potentially including an object (see #6724)
-	if (preg_match('/[OoC]:\+?[0-9]+:"/', $varValue))
+	if (version_compare(PHP_VERSION, '7.0', '>='))
 	{
-		trigger_error('The deserialize() function does not allow serialized objects', E_USER_WARNING);
-
-		return $blnForceArray ? array($varValue) : $varValue;
+		$varUnserialized = @unserialize($varValue, array('allowed_classes' => false));
```

Comment:
Good question.
We could also keep the `preg_match` for PHP 7, and only add the `['allowed_classes' => false]` for extra security.